wellcomecollection/wellcomecollection.org
# Add Meta and TikTok pixels

## What does this change?

#10894 
Adds the Meta and TikTok pixels, as well as dealing with consent before firing them.
We chose not to add the `<noscript>` version for the Meta pixel as we can't verify consent without JS.

> [!NOTE]
> [We've chosen to comment the TikTok script for now](https://wellcome.slack.com/archives/CUA669WHH/p1722248466950899) as we don't have the means to test consent. The ticket won't be closed until that's done, but we're happy to go ahead with Meta for now. Do review the code for TikTok anyway if possible as the next PR will only uncomment its component declaration.

## How to test

- Run locally
- Clear out cookies
- First landing and encounter with cookie preferences banner
  - Confirm scripts have the "hold consent" functions loaded and nothing is sent in Network tab
  - Choose Accept all, check if scripts are updated and see in the network tab if anything is fired. I believe for Meta, the `initator` is `fbevents.js`.
  - Navigate and keep and eye on Network tab
  - Go to footer and click Manage cookies, change Marketing consent to OFF
  - Observe if scripts are removed/updated and if anything fires in the Network tab

[See my comment below](https://github.com/wellcomecollection/wellcomecollection.org/pull/11047#issuecomment-2255555354) for how I tested it in the Meta platform itself.

## How can we measure success?

We track users only when they have given Marketing consent

## Have we considered potential risks?

We track them even if they haven't given Marketing consent.
